### PR TITLE
refactor pec ports initialization in arc

### PR DESCRIPTION
**Pull request by mmandlis**
potentially a follow up PR for https://github.com/PolymerLabs/arcs/pull/3298
WDYT?

### CODE REVIEW

File: shells/pipes-shell-2/pipe.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ const populateDispatcher = (dispatcher, api, composerFactory, storage, context)
       return await addPipeEntity(msg.entity);
     },
     autofill: async (msg, tid, bus) => {
-      return await autofill(msg, tid, bus, composerFactory, storage, context, portFactory);
+      return await autofill(msg, tid, bus, composerFactory, storage, context, [portFactory.bind(null, bus)]);
```
Comment (sjmiles) [inline shells/pipes-shell-2/pipe.js]: I suggest that pec-port export a `portIndustry`, a function that takes a bus and returns a `portFactory`.

File: src/runtime/arc.ts [hunk 2]
```diff
@@ -93,11 +93,11 @@ export class Arc {
...
   readonly pec: ParticleExecutionHost;
 
-  constructor({id, context, pecFactory, slotComposer, loader, storageKey, storageProviderFactory, speculative, innerArc, stub, inspectorFactory, ports} : ArcOptions) {
+constructor({id, context, pecFactories, slotComposer, loader, storageKey, storageProviderFactory, speculative, innerArc, stub, inspectorFactory} : ArcOptions) {
     // TODO: context should not be optional.
     this._context = context || new Manifest({id});
-    // TODO: pecFactory should not be optional. update all callers and fix here.
-    this.pecFactory = pecFactory || FakePecFactory(loader).bind(null);
+    // TODO: pecFactories should not be optional. update all callers and fix here.
+    this.pecFactories = pecFactories && pecFactories.length > 0 ? pecFactories.slice() : [FakePecFactory(loader).bind(null)];
```
Comment (sjmiles) [inline src/runtime/arc.ts]: `FakePecFactory` could also be an industry instead.

File: shells/pipes-shell-2/pipe.js (hunk 1 above)
Comment (mmandlis) [reply shells/pipes-shell-2/pipe.js]: done, thanks!

File: src/runtime/arc.ts (hunk 2 above)
Comment (mmandlis) [reply src/runtime/arc.ts]: i'll send a follow up pr

**Pull request merged**